dotCMS/src/main/java/org/apache/velocity/runtime/resource/ResourceManagerImpl.java

```diff
@@ -287,7 +295,10 @@ public Resource getResource(final String resourceName, final int resourceType, f
          * (static content from #include) with a Template.
          */
 
-        final String resourceKey = resourceType + resourceName;
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
```

Comment:
This is not how this works - we cannot cache a resource per user.

Reply:
The velocity resource and the loaders write out a final generated template as a string and do not have "runtime" access to thread information, etc.  If you include thread or request information in the velocity template it will get cached like that for all time - making it look like it worked first time, but not after. And if you include the user, then it becomes harder to cache.  The loaders are code generators.

Reply:
but we are saving permission info into the html that why we need cache the html by user.
and this is not new, here for example:

https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java#L121

Reply:
"And if you include the user, then it becomes harder to cache", why?

Reply:
The Loaders should be thought of like static code generators and are run once initial template compile time
The java code in the loaders does NOT have access to the runtime/threadlocal when they are called/rendered, except through the use of $Viewtools.  Here is what we need to do.

1. remove the user and the threadlocal from the ContainerLoader, resourceloader
2. Create a new method in the ContainerWebAPI that checks if a user has edit permissions to the container and returns a bool or a string of the bool
3. Write the call to that method as part of the StringWriter that generates the  Container Velocity template.  It is exactly like we do for Contentlets here:
```
https://github.com/dotCMS/core/pull/13729/files#diff-0621778ffa3446ddda7b17d8106804dfR246
```

Reply:
ok